lib/library.rb

```diff
@@ -0,0 +1,40 @@
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books, :times_checked_out
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @times_checked_out = {}
```

Comment:
`times_checked_out` is a great thing to track. I probably wouldn't expose the whole data structure, but make a helper method where we pass in a book object. This would protect our hash from other kinds of methods being called on it that maybe we don't want others to know. (this wasn't asked for, I'm just showing different perspective)